mod-source-record-storage-server/src/test/java/org/folio/services/MarcBibliographicMatchEventHandlerTest.java

```diff
@@ -0,0 +1,412 @@
+      .withSnapshotId(snapshotId1)
+      .withGeneration(0)
+      .withRecordType(MARC)
+      .withRawRecord(rawRecord)
+      .withParsedRecord(new ParsedRecord()
+        .withId(recordId)
+        .withContent(PARSED_CONTENT_WITH_ADDITIONAL_FIELDS))
+      .withExternalIdsHolder(new ExternalIdsHolder()
+        .withInstanceId("681394b4-10d8-4cb1-a618-0f9bd6152119")
+        .withInstanceHrid("12345"));
+
+    SnapshotDaoUtil.save(postgresClientFactory.getQueryExecutor(TENANT_ID), snapshots).onComplete(save -> {
+      if (save.failed()) {
+        context.fail(save.cause());
+      }
+      async.complete();
+    });
+  }
+
+  @After
+  public void cleanUp(TestContext context) {
+    Async async = context.async();
+    SnapshotDaoUtil.deleteAll(postgresClientFactory.getQueryExecutor(TENANT_ID)).onComplete(delete -> {
+      if (delete.failed()) {
+        context.fail(delete.cause());
+      }
+      async.complete();
+    });
+  }
+
+  @Test
+  public void shouldMatchByMatchedIdField(TestContext context) {
+    Async async = context.async();
+
+    WireMock.stubFor(post(PUBSUB_PUBLISH_URL)
+      .willReturn(WireMock.noContent()));
+
+    HashMap<String, String> payloadContext = new HashMap<>();
+    payloadContext.put(MARC_BIBLIOGRAPHIC.value(), Json.encode(record));
+
+    DataImportEventPayload dataImportEventPayload = new DataImportEventPayload()
+      .withContext(payloadContext)
+      .withTenant(TENANT_ID)
+      .withCurrentNode(new ProfileSnapshotWrapper()
+        .withId(UUID.randomUUID().toString())
+        .withContentType(MATCH_PROFILE)
+        .withContent(new MatchProfile()
+          .withExistingRecordType(MARC_BIBLIOGRAPHIC)
+          .withIncomingRecordType(MARC_BIBLIOGRAPHIC)
+          .withMatchDetails(singletonList(new MatchDetail()
+            .withMatchCriterion(EXACTLY_MATCHES)
+            .withExistingMatchExpression(new MatchExpression()
+              .withDataValueType(VALUE_FROM_RECORD)
+              .withFields(Lists.newArrayList(
+                new Field().withLabel("field").withValue("999"),
+                new Field().withLabel("indicator1").withValue("f"),
+                new Field().withLabel("indicator2").withValue("f"),
+                new Field().withLabel("recordSubfield").withValue("s"))))
+            .withExistingRecordType(MARC_BIBLIOGRAPHIC)
+            .withIncomingRecordType(MARC_BIBLIOGRAPHIC)
+            .withIncomingMatchExpression(new MatchExpression()
+              .withDataValueType(VALUE_FROM_RECORD)
+              .withFields(Lists.newArrayList(
+                new Field().withLabel("field").withValue("948"),
+                new Field().withLabel("indicator1").withValue(""),
+                new Field().withLabel("indicator2").withValue(""),
+                new Field().withLabel("recordSubfield").withValue("a"))))))));
+
+    CompletableFuture<DataImportEventPayload> future = new CompletableFuture<>();
+    recordDao.saveRecord(record, TENANT_ID)
```

Comment:
I propose to move call of the method recordDao.saveRecord() to the setUp() method to simplify async coordination code and avoid hanging of test in case saveRecord() returns failed future